src/main/java/org/mskcc/cbio/oncokb/service/UserService.java

```diff
@@ -508,8 +507,24 @@ public void changePassword(String currentClearTextPassword, String newPassword)
     }
 
     @Transactional(readOnly = true)
+    public List<UserDTO> findAllUsersWithUserDetailsByUsersIn(List<User> users) {
+        List<Object[]> usersWithDetails = userRepository.findAllUsersWithUserDetailsByUsersIn(users);
+        return usersWithDetails
+            .stream()
+            .map(u -> {
+                User user = (User) u[0];
+                UserDetails userDetails = (UserDetails) u[1];
+                return userMapper.userToUserDTO(user, userDetails);
+            })
+            .collect(Collectors.toList());
+    }
+
+    @Transactional(readOnly = true)
+    @Cacheable(cacheResolver = "userCacheResolver", key = "#root.methodName")
     public Page<UserDTO> getAllManagedUsers(Pageable pageable) {
```

Comment:
Could you do me a favor checking whether we can get rid of the threshold https://github.com/oncokb/oncokb-public/blob/master/src/main/webapp/app/pages/userManagement/UserDetailsPage.tsx#L78? So the pageable will be null here I assume? If not, let's change that to a bigger size? 10K? @calvinlu3

Reply:
The Pageable object populates with default parameters when we get rid of the threshold. The default size is 20.
I'm increasing the page size to 10k.